test/over_react/util/cast_ui_factory_test.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@TestOn('browser')
+library cast_ui_factory_test;
+
+import 'package:over_react/over_react.dart';
+import 'package:test/test.dart';
+
+part 'cast_ui_factory_test.over_react.g.dart';
+
+main() {
+  group('castUiFactory', () {
+    test('infers the typing of a factory', () {
+      UiFactory<BasicProps> testFactory = castUiFactory(_$Basic);
+
+      expect(testFactory, isA<UiFactory<BasicProps>>());
+    });
+
+    test('can use generics to set the type', () {
+      final testFactory = castUiFactory<UiFactory<BasicProps>>(_$Basic);
+
+      expect(testFactory, isA<UiFactory<BasicProps>>());
+    });
+
+    test('will leave the type a `UiFactory<UiProps` if no type is provided', () {
+      final testFactory = castUiFactory(_$Basic);
+
+      expect(testFactory, isA<UiFactory<UiProps>>());
```

Comment:
These runtime type checks aren't really doing much, especially in this case, since the value of the object will always have the same type.

One way I can think of to assert the static type of these variables would be to perform assignments that would result in a static type error if implicit casts were disabled (which they will be in my PR).

For example:
```suggestion
      // This assignment should analyze properly without any static implicit_cast errors
      UiFactory<UiProps> testAssignment = testFactory;
```

Reply:
Very true. It was a weird case where I felt like tests should exist for a crucial API, but at the same time, static checking is really the way to test it.

Just to be sure, given that we now have a lot of examples using the utility, do you think we need a test suite at all?

Reply:
We probably don't need it, but I think there's value in having unit tests that verify its behavior in these edge-cases, to ensure we understand that behavior fully, as well as to ensure the behavior doesn't change without us knowing about it in a new Dart SDK or something

Reply:
Aand adding this did catch a small issue with implicit casts turned on. So that was perfect!